2017/proposals/yashchuk-ivan.md

```diff
@@ -0,0 +1,120 @@
+# Develop assembly of finite element forms on quadtrilateral and hexahedral meshes
+
+## Abstract
+
+One of the first steps in the finite element method (FEM) is splitting the domain on which the partial differential equation (PDE) is solved into small parts, 
+called cells, which in sum make a mesh. FEniCS has always supported meshes consisting of simplex cells (e.g. triangles and tetrahedrons),
+but has limited support for meshes consisting of quadrilateral (quad) and hexahedral (hex) cells. 
+Finite element problems solved on quad/hex meshes often have better approximation properties and better robustness
+to cell distortion than those solved on simplex meshes.
+The project idea aims at being able to assemble and solve the simplest PDE, a Poisson's equation, in 2D (quad mesh) and 3D (hex mesh) in FEniCS.
+
+## Technical Details
+
+Many constituent parts to assemble and solve on quad/hex meshes are already in FEniCS, but
+there are missing links to get the pieces working as a whole.
+
+A key technological innovation in FEniCS is the development of a domain specific language
+for specifying finite element variational forms (UFL) and a form compiler (FFC) that can 
+translates UFL into low-level C++ code that is used to generate cell tensors (local matrices) on
+every cell in the mesh.
+
+Currently this toolchain cannot produce the C++ code for quad/hex cell geometries. What is needed to do is 
+to interface FFC with an existing class in FInite element Automatic Tabulator (FIAT) to evaluate the tensor product finite 
+element basis functions on the quad/hex cell geometry. An appropriate features
+are needed to be added into FFC to compute geometric quantities on the quad/hex cell geometry, using the existing
+code for triangles and tetrahedrons as an example. Finally, linking everything to the DOLFIN problem solving 
+environment should be done to solve a complete problem.
+
+The main source of information for this project is [The FEniCS Book](https://link.springer.com/book/10.1007%2F978-3-642-23099-8).
+Any other book on FEM can also be useful (for example a [book by Claes Johnson](https://www.amazon.com/Numerical-Solution-Differential-Equations-Mathematics/dp/048646900X))
+
+## Schedule of Deliverables
+
+### May 4th - May 30th, **Community Bonding Period**
+
+* Familiarize myself completely with FEniCS functionality and architecture.
+* Be in constant touch with my mentor and the FEniCS community.
+* Thus with the help of my mentor I will become absolutely clear about my future goals, the final implementations that need to be done as well as the approach that I will follow.
+
+### May 30th - June 3rd
+
+* Start working on building an interface between FFC and FIAT for quad/hex mesh.
```

Comment:
Having this script work would be a good first milestone:
```
from dolfin import *
mesh = UnitQuadMesh(mpi_comm_world(), 1, 1)
print(assemble(1.0*dx(mesh)))
```
Running this today will lead you straight to the place in FFC where the FIAT/FFC interface needs to be implemented for Q elements. Using `UnitHexMesh` instead will allow testing hexahedral cells as well.